server/server.go

```diff
@@ -118,6 +118,7 @@ type Config struct {
 	GRPCServerTimeout                  time.Duration `yaml:"grpc_server_keepalive_timeout"`
 	GRPCServerMinTimeBetweenPings      time.Duration `yaml:"grpc_server_min_time_between_pings"`
 	GRPCServerPingWithoutStreamAllowed bool          `yaml:"grpc_server_ping_without_stream_allowed"`
+	GRPCServerMaxInflightRequests      int           `yaml:"grpc_server_max_inflight_requests"`
```

Comment:
I'm not super convinced of having a global config. I think in practice it's more useful to let the upstream project to have control over it (e.g. limit only specific methods). Have you considered designing so that we just configure a set of 0+ limiters, each limiter being able to limit a configurable set of methods? I'm asking because in Mimir ingester I would like to limit only the push gRPC and not every gRPC call with a "global" limiter.

Reply:
> Have you considered designing so that we just configure a set of 0+ limiters, each limiter being able to limit a configurable set of methods? I'm asking because in Mimir ingester I would like to limit only the push gRPC and not every gRPC call with a "global" limiter.

This PR allows you to do exactly that already. There's global inflight limit, which I think can be useful but can also be disabled, and there's also a possibility to limit specific methods by using `GrpcMethodLimiter`. I imagine that Mimir would install its own `GrpcMethodLimiter` to implement ingesters's inflight limit.

Reply:
I would just remove the global one and let the upstream project to control it via `GrpcMethodLimiter`. WDYT?

Reply:
> I would just remove the global one and let the upstream project to control it via `GrpcMethodLimiter`. WDYT?

OK, let's do that.

Reply:
I've removed support for max inflight requests in gRPC server, and only added limiting via `GrpcInflightMethodLimiter`. Please take a look again.